include/envoy/http/api_listener.h

```diff
@@ -20,14 +20,15 @@ class ApiListener {
    *                         response are backed by the same Stream object.
    * @param is_internally_created indicates if this stream was originated by a
    *   client, or was created by Envoy, by example as part of an internal redirect.
-   * @return StreamDecoder& supplies the decoder callbacks to fire into for stream decoding events.
+   * @return RequestStreamDecoder& supplies the decoder callbacks to fire into for stream
+   *   decoding events.
    */
-  virtual StreamDecoder& newStream(StreamEncoder& response_encoder,
-                                   bool is_internally_created = false) PURE;
+  virtual RequestStreamDecoder& newStream(ResponseStreamEncoder& response_encoder,
```

Comment:
Do you think there's benefit from having Stream in here?

I mildly prefer Http::ResponseEncoder over Http::ResponseStreamEncoder
but your call.

Reply:
Yeah probably not. I can do the rename, but if it's OK with you I would rather do a bulk find/replace after I'm done with my current WIP PR which this change would cause a huge amount of conflicts with. Is that OK?

Reply:
Hm, I don't mind doing the extra review at all, my only concern is if folks use the mocks in their own downstream unit tests, they'd all have to replace both.   If you can land all three in quick succession (send the review for part 2 with a diffbase so I can get it pre-approved?) or if you can find the right sed so it's simple to replace here and in your WIP PR that might be better?

Reply:
Also as said it's optional.  I think I'd take the wordy name over other folks having to do a double migration if it comes down to that :-)

Reply:
Let me do the rename in the PR I'm working on now, and I can post it first against this branch, and then you can decide whether to just merge the other one as a single PR or to do it individually.